### ISSUE TITLE
Concurrent read write on expireNodes from Geth

### ISSUE THREAD

**Issue by terencechain** [prysmaticlabs/prysm#5513]
Prysm node panic'ed due to concurrent read/write on `p2p/enode/nodedb.go:357`

```
[2020-04-19 13:08:00]  INFO blockchain: Executing state transition on block root=0x8225d986... slot=12340
[2020-04-19 13:08:01]  INFO blockchain: Finished applying state transition attestations=11 attesterSlashings=0 deposits=0 proposerSlashings=0 slot=12340 voluntaryExits=0
[2020-04-19 13:08:12]  INFO blockchain: Executing state transition on block root=0xcc57476b... slot=12341
[2020-04-19 13:08:12]  INFO blockchain: Finished applying state transition attestations=6 attesterSlashings=0 deposits=0 proposerSlashings=0 slot=12341 voluntaryExits=0
fatal error: concurrent map read and map write

goroutine 434471 [running]:
runtime.throw(0x174de2a, 0x21)
    GOROOT/src/runtime/panic.go:1112 +0x72 fp=0xc005466d28 sp=0xc005466cf8 pc=0x455e02
runtime.mapaccess1(0x1507f00, 0xc0007a4840, 0xc005466ec0, 0x34d493d8907c7c4f)
    GOROOT/src/runtime/map.go:411 +0x24b fp=0xc005466d68 sp=0xc005466d28 pc=0x42f56b
github.com/ethereum/go-ethereum/p2p/enode.(*DB).expireNodes(0xc001fe2d60)
    external/com_github_ethereum_go_ethereum/p2p/enode/nodedb.go:357 +0x269 fp=0xc005466f28 sp=0xc005466d68 pc=0xeff819
github.com/ethereum/go-ethereum/p2p/enode.(*DB).expirer(0xc001fe2d60)
    external/com_github_ethereum_go_ethereum/p2p/enode/nodedb.go:334 +0x8b fp=0xc005466fd8 sp=0xc005466f28 pc=0xeff4db
runtime.goexit()
    GOROOT/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc005466fe0 sp=0xc005466fd8 pc=0x488ec1
created by github.com/ethereum/go-ethereum/p2p/enode.(*DB).ensureExpirer.func1
    external/com_github_ethereum_go_ethereum/p2p/enode/nodedb.go:323 +0x3e
```

**Comment by terencechain** [comment]
As mentioned. Our geth commit is like 7 months old. Thanks @prestonvanloon for helping to update it

**Comment by prestonvanloon** [comment]
@nisdas We are seeing many use cases of this. We need to priortize this with a patch if #5523 can't go in right away. Please advise.

**Comment by nisdas** [comment]
I will try working on #5523 later today, if not I will release a patch for this in our fork.

**Comment by rauljordan** [comment]
Fixed by #5513
